### PR TITLE
roachtest: anonymize failures of `workload-replay` in roachtest

### PR DESCRIPTION
**Pull request by rickystewart**
Epic: DEVINFHD-809
Release note: None

### CODE REVIEW

**Review by rytaft — APPROVED** [review]
:lgtm: although I'm definitely not an expert in this part of the code

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99718#-:-NRclOoiBP6ZGPRAL93Q:b-7hl3be)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rickystewart, @smg260, and @srosenberg)

___
*[`pkg/cmd/roachtest/github.go` line 179 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/99718#-NRckvB2BUQjG80bzgQT:-NRckvB3-9d_hRp-OoKT:bmptdfh) ([raw file](https://github.com/cockroachdb/cockroach/blob/b5548525164395e6dec4223078826a07492886cd/pkg/cmd/roachtest/github.go#L179)):*
> ```Go
> 
> 	issueMessage := messagePrefix + message
> 	if strings.Contains(issueName, "costfuzz/workload-replay") {
> ```

maybe this should just be `workload-replay`, since I think we plan to use this for more tests than just `costfuzz` going forward.

**Comment by renatolabs** [comment]
Matching on a test name is a little too fragile, in the sense that we could inadvertently rename a test (directly or indirectly), and silently break this logic.

I suggest introducing a new field to `TestSpec` (`Private`, `RedactFailures`, or something like that), and then applying this logic based on the value of that field.

File: pkg/cmd/roachtest/github.go [hunk 1]
```diff
@@ -174,12 +175,16 @@ func (g *githubIssues) createPostRequest(
 		clusterParams[roachtestPrefix("encrypted")] = fmt.Sprintf("%v", g.cluster.encAtRest)
 	}
 
+	issueMessage := messagePrefix + message
+	if strings.Contains(issueName, "workload-replay") {
+		issueMessage = "This details about this test failure have been omitted; consult the log for more details"
```
Comment (renatolabs) [inline pkg/cmd/roachtest/github.go:180]: Nit: `The details`

**Comment by rickystewart** [comment]
> Matching on a test name is a little too fragile, in the sense that we could inadvertently rename a test (directly or indirectly), and silently break this logic.

Well, I think I have no options besides waiting for #95390 to be merged in this case because the test does not actually exist yet.

**Comment by renatolabs** [comment]
Why not create the infrastructure here (creating the field + associated logic), and ensure #95390 rebases and sets the field?

**Comment by rickystewart** [comment]
Fine. @rytaft and @rachitgsrivastava, please keep in mind you must set `RedactResults` for the new test we're adding.

**Review by renatolabs — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99718#-:-NRe1gAqBsgfmBvQBmQo:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @rickystewart, @rytaft, @smg260, and @srosenberg)

**Comment by rickystewart** [comment]
bors r=rytaft,renatolabs

**Pull request merged**